src/python/pants/backend/pants_info/explain_options.py

```diff
@@ -190,3 +220,32 @@ def console_output(self, targets):
                         inner_map["history"] = history_list
         if self.is_json():
             yield json.dumps(output_map, indent=2, sort_keys=True, cls=CoercingOptionEncoder)
+
+
+@goal_rule
+def explain_options(
+    explain_options_options: ExplainOptionsOptions,
+    options_wrapper: _Options,
```

Comment:
This isn't a public type. It would be much cleaner to do this via the `HelpPrinter`, which already implements most of this same logic, but additionally needs to consume graph edges.

Reply:
I will create an issue for that, but for now I would like to get this in as-is, because it unblocks the downstream refactors that will make changing HelpPrinter easier (originally to support generating the options reference, but now also this).  

Also TBH I'm not 100% convinced on the merits - the HelpPrinter is already clunky, and adding more modes and options to it is going to make it even clunkier. E.g., where would the options currently on `ExplainOptionsOptions` live?  I think the less special-casing the better, and since this is clearly easy to implement as a rule, maybe that's the way to go? 

Concretely, what is the downside to allowing rules to request _Options? It might be fine to have non-public types that are only for internal use.

Reply:
>  I think the less special-casing the better, and since this is clearly easy to implement as a rule, maybe that's the way to go?

It does so by using the mutable private fields ("tracker") of a private type. The fact that `_Options` is exposed at all as an intermediate type between `ScopedOptions` and the `OptionsBootstrapper` is an implementation detail that would be good to remove.

> E.g., where would the options currently on ExplainOptionsOptions live?

They could live on a subsystem.

> Concretely, what is the downside to allowing rules to request _Options? It might be fine to have non-public types that are only for internal use.

There are not enough APIs exposed to `@rules` to allow for implementing the graph awareness that is necessary to implement "help for all reachable types" as described in #8884 (and which is needed for the docsite generation). And unless we have a desperate need for allowing rules to introspect the rule graph like that, we should not add APIs to do it. `UnionMembership` has proven to be necessary, but it is intentionally quite limited.

Rather than designing an API to wrap an API on the scheduler for use in exactly one place in a rule, I think that we should consume that API outside the engine.

Reply:
> I will create an issue for that, but for now I would like to get this in as-is, because it unblocks the downstream refactors that will make changing HelpPrinter easier (originally to support generating the options reference, but now also this).

If there are upcoming changes to move this back out into the `HelpPrinter` (which would be necessary to solve #8884, I think), then landing this would be ok.

Reply:
Well, I'm agreeing that the docsite generation can't be a rule, given this reasoning. I wasn't as sure about this thing, but I accept your argument there too. But regardless, I'm in a circular problem right now because I need some solution for `options` in order to keep that test passing, so I can get my yak shave unwound, so I can make the necessary changes to HelpPrinter for both the docsite generation and now this. It might be possible to do that without the yak shave, but it'll be more annoying. So if possible I would like to get this in temporarily (or alternatively, xfail the test temporarily...)